src/app/clusters/time-format-localization-server/time-format-localization-server.cpp

```diff
@@ -111,30 +108,23 @@ CHIP_ERROR TimeFormatLocalizationAttrAccess::Read(const ConcreteReadAttributePat
 bool IsSupportedCalendarType(CalendarTypeEnum newType, CalendarTypeEnum & validType)
 {
     // Reset valid type if no supported calendar types found.
-    validType = CalendarTypeEnum::kBuddhist;
+    validType = CalendarTypeEnum::kUseActiveLocale;
+    if (newType == CalendarTypeEnum::kUseActiveLocale)
+    {
+        return true;
+    }
 
     DeviceLayer::DeviceInfoProvider * provider = DeviceLayer::GetDeviceInfoProvider();
+    VerifyOrReturnValue(provider != nullptr, false);
 
-    if (provider)
-    {
-        DeviceLayer::DeviceInfoProvider::SupportedCalendarTypesIterator * it = provider->IterateSupportedCalendarTypes();
+    AutoReleaseIterator it(provider->IterateSupportedCalendarTypes());
+    VerifyOrReturnValue(it.IsValid(), false);
 
-        if (it)
+    while (it.Next(validType))
+    {
+        if (validType == newType)
```

Comment:
This is broken; doesn't do what it used to do.

Stopping here.  Please check over things carefully

Reply:
reverted the temporary variable usage. I assume this was done to allow Next to modify the input even when returning false ... although that to me seems a bit broken.

Reply:
Reverted all this refactoring business into #30556